pkg/localstorev2/internal/chunkstore/txchunkstore_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chunkstore_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/localstorev2/internal/chunkstore"
+	postagetesting "github.com/ethersphere/bee/pkg/postage/testing"
+	"github.com/ethersphere/bee/pkg/sharky"
+	chunktest "github.com/ethersphere/bee/pkg/storage/testing"
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/storagev2/inmemstore"
+	"github.com/ethersphere/bee/pkg/storagev2/storagetest"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/spf13/afero"
+)
+
+func TestTxChunkStore(t *testing.T) {
+	t.Parallel()
+
+	sharky, err := sharky.New(&memFS{Fs: afero.NewMemMapFs()}, 1, swarm.SocMaxChunkSize)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	storagetest.TestTxChunkStore(t, chunkstore.NewTxChunkStore(inmemstore.NewTxStore(inmemstore.New()), sharky))
+}
+
+// TestMultipleStampsRefCnt tests the behaviour of ref counting along with multiple
+// stamps to ensure transactions work correctly.
+func TestMultipleStampsRefCnt(t *testing.T) {
+	t.Parallel()
+
+	sharky, err := sharky.New(&memFS{Fs: afero.NewMemMapFs()}, 1, swarm.SocMaxChunkSize)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	chunkStore := chunkstore.NewTxChunkStore(inmemstore.NewTxStore(inmemstore.New()), sharky)
+
+	chunk := chunktest.GenerateTestRandomChunk()
+	stamps := []swarm.Stamp{chunk.Stamp()}
+	for i := 0; i < 2; i++ {
+		stamps = append(stamps, postagetesting.MustNewStamp())
+	}
+
+	verify := func(t *testing.T) {
```

Comment:
nit-picks:
- could `verify` function have more descriptive name, e.g. `assertStoreContainsStamps`?
- could this function be defined outside `TestMultipleStampsRefCnt` so it can be potentially reused?